topic_monitor/topic_monitor/scripts/topic_monitor.py

```diff
@@ -306,7 +306,10 @@ def run_topic_listening(node, topic_monitor, options):
         # TODO(dhood): use graph events rather than polling
         topic_names_and_types = node.get_topic_names_and_types()
 
-        for topic_name, type_name in topic_names_and_types:
+        for topic_name, type_names in topic_names_and_types:
+            if len(type_names) != 1:
+                continue
```

Comment:
I recommend putting a message here, otherwise there is no feedback to the user that a topic it is monitoring is being ignored.

In the bridge I kept a list of topic names for which a similar warning was printed and used it to only print once.